src/Rotations.jl

```diff
@@ -41,6 +41,12 @@ inner(M::Rotations, x, w, v) = dot(w, v)
 
 norm(M::Rotations, x, v) = norm(v)
 
+function project_point!(M::Rotations, y, x)
```

Comment:
can you add a small doctoring here mentioning the method of projection? That would be nice, then it shows um in the docs.

Reply:
And BTW, this function is incorrect. It projects onto the orthogonal group (rotations and reflections). This causes some tests to fail.

I don't know if there is a consistent way to project onto SO(n), I've always just used an SVD or QR (QR is generally faster) with an arbitrary reflection if needed. Like here: https://github.com/mateuszbaran/FunManifolds.jl/blob/04543dc52784762016bf7090e2fcfad5f98dd172/src/special_orthogonal.jl#L261

Reply:
Yes, thanks, the change to the test helped me realize that. I saw a number of references that used the SVD approach with a sign change to the vector corresponding to the smallest singular value if necessary. It's not entirely clear from the references I saw, but it sounds like this is more reliable than the QR approach.

Ultimately, it might be good to think of an interface for specifying projection methods. e.g. on `Sphere`, if you know the point should be very close to the sphere and are only projecting to correct minor numerical imprecision, then you can do `x .*= 2 / (1 + dot(x, x))` and forego the square root. And for `Rotations`, one could (maybe) use QR instead of SVD and forego the reflection.

Reply:
My numerical experiments indicate that indeed the SVD approach always gives an element of SO(n) closer to the original point than the QR approach does, under the Frobenius norm.

Reply:
One can get back from O(n) to SO(n) by dividing by the sign of the determinant, since all matrices in SO(n) have determinant +1.

Reply:
Yes, exactly, the SVD approach returns the closest rotation in the Frobenius norm but QR is faster. We need an `AbstractProjectionMethod`.

> One can get back from O(n) to SO(n) by dividing by the sign of the determinant, since all matrices in SO(n) have determinant +1.

Dividing what? Dividing/multiplying a 2k by 2k matrix by -1 doesn't change the determinant.

Reply:
Oh, indeed, I was a little short in thinking: Why changing all signs, one sign should be enough – but yes that of the determinant does not help here for even cases ...